### ISSUE TITLE
Do not collect pod metrics using Bottlerocket

### ISSUE THREAD

**Issue by SpringMT** [aws-observability/aws-otel-collector#1071]
**Describe the bug**
When using Bottlerocket in EKS, otel-collector can not collect Pod metrics.
Log is bellow.

```
2022-03-20T14:04:42.986Z	warn	cadvisor/container_info_processor.go:93	No pod metric collected	{"kind": "receiver", "name": "awscontainerinsightreceiver", "metrics count": 27}
```

**Steps to reproduce**
1. Create cluster with Bottlerocket
2. Set aws-otel-collector with `aws-otel-collector/deployment-template/eks/otel-container-insights-infra.yaml`
  * I use `amazon/aws-otel-collector:v0.17.0` image

**What did you expect to see?**
This bug is similar to this issue https://github.com/aws/amazon-cloudwatch-agent/issues/188.
This issue fixed by https://github.com/aws/amazon-cloudwatch-agent/pull/189/.

`aws-otel-collector` uses container_info_processor in opentelemetry-collector-contrib.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/awscontainerinsightreceiver/internal/cadvisor/container_info_processor.go

I think the similar changes in https://github.com/aws/amazon-cloudwatch-agent/pull/189/  will fix this bug.

**What did you see instead?**

**Environment**
Bottlerocket in EKS

**Additional context**

**Comment by mhausenblas** [comment]
@SpringMT can you please confirm that this still happens with v0.18 of the ADOT collector?

**Comment by YDKK** [comment]
@mhausenblas I have same issue and have confirmed that this still happen with the `amazon/aws-otel-collector:v0.18.0` image.

**Comment by YDKK** [comment]
As a workaround, I created a VolumeMount from `/run/dockershim.sock` to `/run/containerd/containerd.sock` (the same way as CloudWatch Agent) and it works fine.

**Comment by mmomont** [comment]
@YDKK do you mind sharing:
`kubectl describe ds adot-collector-daemonset` ?

Once I updated the volume I still do not see metrics and a warn message appears:
`2022-06-10T16:26:13.017Z	warn	cwlogs@v0.45.1/cwlog_client.go:84	cwlog_client: Error occurs in PutLogEvents, will search the next token and retry the request	{"kind": "exporter", "name": "awsemf", "error": "InvalidSequenceTokenException: The given sequenceToken is invalid. The next expected sequenceToken is: 49629762964284285237968457586128106115364986877335006674\n{\n  RespMetadata: {\n    StatusCode: 400,\n    RequestID: \"127167c6-480e-4139-b63b-bf4c4f0f5fb1\"\n  },\n  ExpectedSequenceToken: \"49629762964284285237968457586128106115364986877335006674\",\n  Message_: \"The given sequenceToken is invalid. The next expected sequenceToken is: 49629762964284285237968457586128106115364986877335006674\"\n}"}`

**Comment by vasireddy99** [comment]
Adding context here:

Interesting that bottle rocket has fixed the mount point to `containerd` for [kubernetes-1.23] support (https://github.com/bottlerocket-os/bottlerocket/blob/cbfbd05f2909eefb63f19c1c536050de2853fb29/packages/kubernetes-1.23/kubelet-exec-start-conf#L15-L16) so if using latest version of bottle rocket mount needs to be fixed to  `/run/containerd/containerd.sock`.

We are currently working on this issue and shall provide more context as soon as we have an update on resolving.

**Comment by vasireddy99** [comment]
Thank you so much for creating this issue and appreciate for the good detail. The Support to collect pod metrics while running on EKS cluster with `containerd` runtime is available from ADOT collector `v0.21`. Closing this issue.

**Comment by SpringMT** [comment]
Thank you!
